frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderView.tsx

```diff
@@ -112,56 +117,67 @@ export function DashboardHeaderComponent({
           <span>{editWarning}</span>
         </EditWarning>
       )}
-      <div>
+      <HeaderContainer isSidebarOpen={isSidebarOpen}>
```

Comment:
@kdoh Adjust dashboard header position to align with dashboard body positioning when dashboard sidebar is open

https://github.com/metabase/metabase/assets/22608765/41706729-ddf5-4ce3-9807-f42a88f41a6f

Slack Discussion: https://metaboat.slack.com/archives/C064QMXEV9N/p1707761462561239?thread_ts=1707503415.376959&cid=C064QMXEV9N